.github/workflows/test.yml

```
@@ -0,0 +1,40 @@
+name: Run Tests
+on:
+  push:
+    branches: [ main, dev ]
+  pull_request:
+    branches: [ main, dev ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.10', '3.11', '3.12']
+
+    services:
+      redis:
+        image: redis:6-alpine
+        ports:
+          - 6379:6379
+
+    steps:
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

```suggestion
        uses: actions/setup-python@v5
```
any reason not to use v5 here?

---

Nope, but I'm happy to update!